pieces.py

```diff
@@ -10,28 +10,52 @@ class Pawn(Piece):
 	def __init__(self, is_white, square):
 		super().__init__(is_white, square)
 		self.movement_type = ("forward", "backward", "forward_left", "forward_right", "backward_left", "backward_right")
```

Comment:
I know this wasn't touched in this branch, but want to mention it anyway:

- Why is `movement_type` a property that is being set on each instance in the constructor? Couldn't this be defined by a method on each class with that class's allowed movement types?
- What are the semantics of these movement types? I notice, for example, that both a pawn and a queen have `forward` movement type, but for the pawn that means only forward 1 (or two) and for the queen it means forward any number of times. Where is that difference being handled and should it be represented here in the movement types? (Maybe what I'm asking is: is `movement_type` a useful concept if you still have helpers like `get_pawn_moves` that just ignore it and do their own per-piece-type logic?)

Reply:
I think movement_type is useful but the code for pawns specifically should be refactored. Pawns just have too many exceptions to fit into the logic that every other piece uses. Even with kings which can only move one space, it's very easy to still use the same logic. But with pawns, they're the only piece that doesn't move backwards. They can move forward 2 spaces, sometimes. They can move forward unless they're capturing. They can move diagonally but only when capturing. Not to mention en passant and promotion. Movement_type basically governs everything but pawns and knights which is still a pretty healthy amount of piece logic.